locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/networking/SecurityGroupEditor.java

```diff
@@ -156,11 +162,14 @@ public AmbiguousGroupName(String s) {
      * @throws AmbiguousGroupName in the unexpected case that the cloud returns more than one matching group.
      */
     public Optional<SecurityGroup> findSecurityGroupByName(final String name) {
-        final String query = name.startsWith(JCLOUDS_PREFIX) ? name : JCLOUDS_PREFIX + name;
         final Iterable<SecurityGroup> groupsMatching = findSecurityGroupsMatching(new Predicate<SecurityGroup>() {
+            final String rawName = name.matches(IS_JCLOUDS_PREFIXED_REGEX)
+                ? name.substring(JCLOUDS_PREFIX_REGEX.length())
+                : name;
+
             @Override
             public boolean apply(final SecurityGroup input) {
-                return input.getName().equals(query);
+                return input.getName().matches(JCLOUDS_PREFIX_REGEX + rawName);
```

Comment:
Need to escape the name with "Pattern.quote(rawName)". Or better yet use the same method as above to strip the prefix and compare the non-prefixed names.